src/common/procedure/src/procedure.rs

```diff
@@ -174,6 +215,54 @@ impl<T: Procedure + ?Sized> Procedure for Box<T> {
     fn lock_key(&self) -> LockKey {
         (**self).lock_key()
     }
+
+    fn poison_keys(&self) -> PoisonKeys {
+        (**self).poison_keys()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct PoisonKey(String);
+
+impl Display for PoisonKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl PoisonKey {
+    /// Creates a new [PoisonKey] from a [String].
+    pub fn new(key: impl Into<String>) -> Self {
+        Self(key.into())
+    }
+}
+
+/// A collection of [PoisonKey]s.
+///
+/// This type is used to represent the keys that may cause the procedure to become poisoned during execution.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, Default)]
+pub struct PoisonKeys(SmallVec<[PoisonKey; 2]>);
+
+impl PoisonKeys {
+    /// Creates a new [PoisonKeys] from a [ResourceType] and a [String].
```

Comment:
```suggestion
    /// Creates a new [PoisonKeys] from a [String].
```

Reply:
resolved in [776d5ec](https://github.com/GreptimeTeam/greptimedb/pull/5822/commits/776d5ec6476243978feb1e84a2546846e7592d38)